### PR TITLE
#1497 fixes msg around assignments are graded and all goals are set

### LINKED ISSUE #1497: Assignment Planning: Display the course goal as null when all assignments are graded and students have not set a goal
### Expected behavior (A description of what you expected to happen) : 
The Minimum Goal for the course should be null and not show an error when a student hasn't set a course and assignment goals, even when all assignments are submitted and graded.

### Describe the bug (Tell us what happens instead of the expected behavior) :
Prerequisites: 

- Student has submitted all assignments on the course
- All assignments have been graded
- Student has not set up a Minimum Goal for the course

In these circumstances, MyLA is displaying the message "Scores no longer match goal" in the My Minimum Goal field.  This field should be blank because the student hasn't set a goal.

### Screenshots/Video (If possible, add media to help explain your problem) :
image

--------------------
Test Plan:
https://docs.google.com/spreadsheets/d/1fOLY2byRBUZ1JzCFir6WiIPfrMPN2_-7oJKr5zhsiEM/edit#gid=0

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #1497 

Updated the test plan: https://docs.google.com/spreadsheets/d/1fOLY2byRBUZ1JzCFir6WiIPfrMPN2_-7oJKr5zhsiEM/edit#gid=2106880085

The alert message logic is around when assignments grades (all ,few, none) and assignment goal grade set by user (all , few none)

### CODE REVIEW

File: assets/src/hooks/useMathWarning.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
 import { useEffect, useState } from 'react'
 
-const allAssignmentsHasGoalGrade = assignments =>
-  assignments.filter(a => !(a.goalGradeSetByUser)).length === 0
+// this is the case when either few assignments are graded or few has goal grade.
+// if an assignment is graded then it won't have goal grade and vice versa.
```
Comment (zqian) [inline assets/src/hooks/useMathWarning.js:4]: Would you put the 'vice versa' scenario into the comment line, too?

File: assets/src/hooks/useMathWarning.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+const isFewAssignmentsAreGradedOrHasGoalGrade = assignments =>
+  assignments.filter(a => !(a.graded || a.goalGradeSetByUser)).length === 0
+
+// this is the case when all assignments are graded and design will won't assignment goal setting but course grade is allowed
```
Comment (zqian) [inline assets/src/hooks/useMathWarning.js:8]: typo "will won't"

Comment (pushyamig) [reply assets/src/hooks/useMathWarning.js]: DOne

File: assets/src/hooks/useMathWarning.js (hunk 1 above)
Comment (pushyamig) [reply assets/src/hooks/useMathWarning.js]: Done

**Review by zqian — APPROVED** [review]

**Pull request merged**